### DISCUSSION TITLE
Verbindung zu Fox Ess H3

### DISCUSSION THREAD

**Discussion by Ingo-Vetter** [nathanmarlor/foxess_modbus#604] [Support] [answered]
Hallo Nathanmarlor,

ich habe zwei Fox H3, die ich in HA integrieren möchte.
Leider bekomme ich keine Verbindung zu den Geräten.
 
Die WR sind mit den mitgelieferten Anatel Wifi Sticks mit dem Netzwerk verbunden.
Dieser Steckt im Anschluss 7 (siehe Bild Fox H3 )

![Anatel Wifi Stick](https://github.com/nathanmarlor/foxess_modbus/assets/26342084/a2256fe1-2d3e-4e66-b4f5-6a038879e8be)
![Fox H3 Anschlüsse](https://github.com/nathanmarlor/foxess_modbus/assets/26342084/594f4cb0-0136-4acf-9559-d2f3367f4c2c)
 
Können Sie mir helfen, die WR zu Integrieren? 

Viele Grüße Ingo

**Comment by FozzieUK** [comment, marked as answer]
The WiFi sticks (dataloggers) are used to connect the inverter to the Foxesscloud, they cannot be used for anything else - this integration needs a direct RS485 modbus connection to work. 

You can see details of some RS485 devices and how to wire them here in the [wiki](https://github.com/nathanmarlor/foxess_modbus/wiki/Modbus-Wiring-Guide)

If you have connected your inverters to the Foxesscloud, you can use the [cloud integration ](https://github.com/macxq/foxess-ha) to gather sensor data but it is limited to read only functions, whereas the modbus integration can control the inverters locally.

**Reply by Ingo-Vetter** [reply]
Ich habe 2 H3.
Diese sind als Master und Slave in einem Verbund.
Der Master ist über die RS485 Schnittstelle mit Smart Meter verbunden.

![Verbindung Smart Meter Stecker 1](https://github.com/nathanmarlor/foxess_modbus/assets/26342084/9c4dc2bc-33fa-4a50-b253-7421edde4c0a)

Kann die RS485 Schnittstelle dann gleichzeitig für ModBus Verbindung genutzt werden, oder muss/kann ich über eine andere Schnittstelle/ einen anderen Weg eine Verbindung zu den Wechselrichtern herstellen?

**Reply by FozzieUK** [reply]
This meter is connected over RS485 directly to the 'meter' input of the inverter, they cannot be used for the modbus connection for this integration, only to advise the inverter the power usage and this is shared between master and slaves over canbus (or parallel as Foxess refer to it).

There are multiple connections within that same connector, one for meter, and another for 'logger', the 'logger' connection is where you would connect the 2 data wires and then link them to your preferred modbus device that this integration addresses.

![1111](https://github.com/nathanmarlor/foxess_modbus/assets/113460294/4bc9155b-0187-499e-87e9-d1c925b80164)

**Reply by Ingo-Vetter** [reply]
Okay, das habe ich jetzt verstanden. 
Dankeschön.
Muss ich jeden WR einzeln einbinden, oder nur den Master?

**Reply by FozzieUK** [reply]
You need to integrate them individually, you can daisy chain the RS485 together onto a single modbus converter and set different slave id’s for each so they can be addressed independently.

**Reply by Ingo-Vetter** [reply]
Ich habe jetzt verstanden, dass ich beide WR mit einem Converter verbinden kann.
Hierzu habe ich mir diesen Converter bestellt.

![USR-TCP232-304 1 Port RS485 TCPIP](https://github.com/nathanmarlor/foxess_modbus/assets/26342084/30068399-6d36-4f65-b61c-78172dccc29f)

Ich verbinde also jeweils A mit 1, B mit 2 und G mit 5 oder 6.
Die Device IDs der WR habe ich auf 1 (Master) und 2 (Slave) eingestellt.
So sollte ich jetzt beider WR mit einer Integration einbinden, richtig?

Wie muss ich vorgehen, wenn ich für jeden WR einen Converter nutzen möchte?

**Reply by FozzieUK** [reply]
For most applications you only need to connect 2 wires (A&B) as they are differential inputs, so yes A to 1, and B to 2.

The TCP232-304 device will work well for 1 H3, but this device doesn’t support modbus it may struggle running 2 together on a daisy chain - I think the best approach would be to get one working with this integration first, and make sure you are reliably receiving data before you consider adding the second. 

If you decide to have one converter on each H3, you would end up with 2 IP addresses, so you would add a second device to the integration.

**Reply by canton7** [reply]
> The TCP232-304 device will work well for 1 H3, but this device doesn’t support modbus it may struggle running 2 together on a daisy chain

I think it will be OK, as long as you don't try to set up more than one integration which uses the TCP232-304 at the same time. With foxess_modbus, you can configure multiple inverters using the same TCP232-304, and it will make sure that it only tries to talk to one inverter at a time.

That said, the [USR-TCP232-410s](https://github.com/nathanmarlor/foxess_modbus/wiki/USR-TCP232-410s) does support modbus, and is a better device to buy for this.

**Reply by Ingo-Vetter** [reply]
Ich habe jetzt USR-TCP232-410s eingebaut und es funktioniert.
Danke

**Comment by horstvogel** [comment]
Moin Ingo,
wie hast Du Dein Dashboard eingestellt, bei mir kommen da für den Zuhauseverbrauch Werte raus, die für mich nicht plausibel sind. Wechselrichter gibt ja nur DC raus, da wird es da Abweichungen geben, aber irgendwie kommt mir das alles "spanisch" vor.

**Reply by Ingo-Vetter** [reply]
Tach auch,
Ich habe bei mir für alle je einen Shelly pro 3em. Also der Haupzähler, die beiden PV Anlagen, die Wärmepumpe und die Garage. 
Hier habe ich sehr gut alle Verbräuche und der Shelly am Zähler zeichnet auf, was rein und rausgeht.

Ja, die Daten aus den Wechselrichtern muss man erstmal verstehen. Haber, es ist auch kein Hexenwerk. 
Ich habe habe die Werte gefunden, die den Verbrauch uns den Zukauf entsprechen, bzw. sehr nahe kommen.

Ich lasse mir in verschiedene Daten Anzeigen.
Im Anhang findest du ein paar Beispiele.

**Reply by Ingo-Vetter** [reply]
![image](https://github.com/user-attachments/assets/debdf808-1fdf-4c6c-9338-5993d6ef41a9)

**Reply by Ingo-Vetter** [reply]
![image](https://github.com/user-attachments/assets/bbb3c278-8d24-4680-b330-80a6065ceda8)

**Reply by Ingo-Vetter** [reply]
![image](https://github.com/user-attachments/assets/3dd67fd1-7b3d-4d2d-9003-c9ad257aaf9b)

**Comment by Ingo-Vetter** [comment]
Ach ja, in DC gibt es nur die Spannungsdaten der Strings. der Rest ist alles AC (Wechselstrom).

**Comment by horstvogel** [comment]
Sorry ich meinte natürlich das Energy Dashboard von HA direkt, da kann man ja leider keinen Messwert für Hausverbrauch einfügen, daher gehe ich davon aus, dass die Werte einfach falsch sind. Alleine schon da Solar Generation und Battery Charge  DC kWh sein müssen. 



Das ist was ich so denke

Yield Total ist was durch den Wechselrichter gegangen ist, egal ob direkt von der PV oder vom Speicher. 
Solar Generation sind die DC kWh vom Dach
Load Energy ist der "Zuhause" Verbrauch AC
Input Energy Total ...keine Ahnung was das ist.
Grid Consumption Total was aus dem Netz bezogen worden ist AC
Feed in Total, dass was ins Netz eingespeist worden ist.  AC
Battery Discharge Total was die Battery als??? DC oder AC entladen hat
Battery Charge das was an DC in die Batterie gegangen ist. 






![image](https://github.com/user-attachments/assets/b383b637-f489-4691-87ce-13b6b1b70c08)

**Reply by Ingo-Vetter** [reply]
Ja guck, da hast du doch schon fast alles gefunden.
Und ja, alles was mit der Batterie zutun hat ist natürlich auch in DC, da diese ja über DC läuft.

Battery Discharge Total ist die Summe von dem was aus der Batterie entnommen wurde.

Da gibt es auch einen Messwert (e) vom SmartMeter. Glaube ich zumindest.
Die Daten passten ganz gut zu den Daten aus dem Shelly vom Hauptzähler.

mich muss nachher mal gucken wie der heißt.

**Comment by horstvogel** [comment]
Und was auch überraschend ist, dass Foxess immer so um die 35 Watt Netzbezug hat. Andere Hybride speisen vermutlich eher auch was ein...

**Reply by Ingo-Vetter** [reply]
Jap, und dass pro WR. 
Und der Verbrauch wird bei Fox nicht mit angezeigt 🤔🫣

Wie ist das bei dir in der Nacht, hast du da Fremdbezug (Zukauf)?

Gruss Ingo

**Reply by horstvogel** [reply]
Das ist ja genau was ich meinte, der Wechselrichter (hier nur einer ein H3) bezieht trotz noch gefüllter Batterie immer so ca. 35 Watt, erst wenn die 100 % erreicht sind, dann wird nichts mehr aus dem Netz bezogen, da dann natürlich der Überschuss. vorhanden ist. 

Pro Tag sind das zwischen 0,4 kWh und 0,7 kWh


Die Batterie war gestern z.B. nicht unter 70 %. Natürlich gibt es Momente wo dann die Hausleitung über den 6 kW des Hybriden ist, aber in der Nacht halt konstant die 35 Watt Bezug, obwohl Speicher "voll"

![image](https://github.com/user-attachments/assets/d7b70fa3-9543-471c-8091-ef6d69c20496)

**Reply by horstvogel** [reply]
Die 13,3 sind der Hausverbrauch laut dem Energie Dashboard, dass ist halt dem geschuldet, dass das Energiedashboard den gemessenen Hausverbrauch nicht kennt. 

Das ganze dann mit der Energy Flow Card Plus. Da kann man den gemessenen Verbrauch von der Foxess Modbus Integration einfügen, wobei sich mir da die Frage stellt, ist der gemessen oder wird der nur berechnet? 

13,3 zu 10,9 kWh. Das ist dann schon ein größerer Unterschied. 

![image](https://github.com/user-attachments/assets/9dd1a949-060c-42f2-9993-6070decf5f1c)

**Reply by horstvogel** [reply]
In der Energy Flow Card Plus geht die Summe dann natürlich nicht auf. Sprich die Summe ist nicht Null. Aber der Hausverbrauch ist da vermutlich richtig, wobei die Frage ja ist "wie errechnet sich  "Load Energy Total" oder wird das gemessen.

**Reply by DerRonnyUK** [reply]
habt ihr eine CT Klemme wo ihr die Daten bzgl hausverbrauch evtl abgreifen koennt? Ansonsten wird ein WR immer eine minimalen "Verbrauch" haben, dieser ist ja reaktiv - im sinne der reagiert auf schwankungen im Hausverbrauch. Du kannst auf feed in prio versuchen umzustellen und schauen ob dann der verbrauch auf 0 geht

**Comment by Ingo-Vetter** [comment]
Guten Morgen,
Ich habe gerade mal bei mir nachgesehen. Die Entität Load Energy Total nutze ich nicht. Ich kann dir auch nicht sagen, was das für ein Wert sein soll. Evtl. die Energie, die in die Batterie gegangen ist? 

Der zugekaufte Strom ist es meines Erachtens nach aber nicht.
Das müsste die Entität Grid Consumption sein.

Weißt du was die Entität Yield für einen Wert anzeigt?

**Reply by horstvogel** [reply]
Load Energy ist der Hausverbrauch. 

Yield Total ist was alles so durch den Wechselrichter gegangen ist. 

so würde ich das verstehen.

**Reply by DerRonnyUK** [reply]
Yield total sollte der gesamte Erzeugte Strom sein?

**Reply by horstvogel** [reply]
Nein wie ich geschrieben habe, würde ich das als die Wechselrichterarbeit verstehen, die durch den Wechselrichter gegangen ist, Egal ob direkt von der Sonne oder über den Speicher. Vielleicht auch AC über den Speicher wieder rein...
Genau das ist halt leider die Frage. Schön war auch noch, wenn man wüsste was ist ein gemessener Wert, was ist dann von der Integration errechneter Wert. 
da ich selbst über keinen Wechselrichter verfüge, kann ich das z.B. nicht mit der Anzeige am Wechselrichter vergleichen. Weiter ist vor Ort bei der Anlage auch nur ein Netzwerkanschluss zur Zeit verfügbar, Da die Einbausituation das so leider nicht hergibt.... 
daher kann das vielleicht von Euch jemand mal mit der Foxess App abgleichen. 

Aber wie der RonnyUK schon einer andern Diskussion geschrieben hat, wäre sehr cool, wenn die einzelnen Werte erklärt werden könnten. 

https://github.com/macxq/foxess-ha/releases/tag/v0.41

Das muss man auch noch verstehen....
https://github.com/macxq/foxess-ha/wiki/Understand-PV-string-power-generation-using-foxess-ha

**Comment by horstvogel** [comment]
https://github.com/nathanmarlor/foxess_modbus/discussions/462
Load Energy hatte ich von da abgeleitet. Muss aber passen, ob das stimmt.
